source/Calamari/Scripts/AzureContext.ps1

```diff
@@ -60,58 +60,109 @@ function Execute-WithRetry([ScriptBlock] $command) {
     }
 }
 
+function Get-AzureRmModuleInstalled {
+    return $null -ne (Get-Command "Login-AzureRmAccount" -ErrorAction SilentlyContinue)
+}
+
+function Get-AzModuleInstalled {
+    return $null -ne (Get-Command "Connect-AzAccount" -ErrorAction SilentlyContinue)
+}
+
+function Get-RunningInPowershellCore {
+    return $PSVersionTable.PSVersion.Major -gt 5
+}
+
+function Initialize-AzureRmContext {
+    
+    # Authenticate via Service Principal
+    $securePassword = ConvertTo-SecureString $OctopusAzureADPassword -AsPlainText -Force
+    $creds = New-Object System.Management.Automation.PSCredential ($OctopusAzureADClientId, $securePassword)
+
+    # Turn off context autosave, as this will make all authentication occur in memory, and isolate each session from the context changes in other sessions
+    Write-Host "##octopus[stdout-verbose]"
+    Disable-AzureRMContextAutosave -Scope Process
+    Write-Host "##octopus[stdout-default]"
+
+    $AzureEnvironment = Get-AzureRmEnvironment -Name $OctopusAzureEnvironment
+    if (!$AzureEnvironment)
+    {
+        Write-Error "No Azure environment could be matched given the name $OctopusAzureEnvironment"
+        exit -2
```

Comment:
It would be nice to exit with positive numbers so it does not overflow in bash

Reply:
Ah yes, I need to check that still, thanks!

Reply:
This is done